Test: Write test ensuring leading 0's for zip code are not removed

jakeleboeuf opened the pull request:
# Describe your PR
We want to ensure any zip with a leading zero doesn't get stripped before sending to the Google API. So `02215` should stay `02215` and not become `2215` when sent to search.

To do this, I'm asserting 2 things:
1) The `XHR Request` to the Google Geocode API
2) The redirect url contains `location=<LEADING_ZERO_ZIP>`

Fixes #215

## Pages/Interfaces that will change
- /businesses (Tests only)
- /allies (Tests only)

### Screenshots / video of changes
https://www.loom.com/share/c8653ef66970469b83fcdc390ea11059

## Steps to test

1. `$ npm run test:e2e`
2. Look for ✅ 

### Additional notes
Zip code search is not currently implemented for the `/allies` page, so those tests are not included in this PR

jakeleboeuf commented on cypress.json:
```diff
@@ -1,7 +1,4 @@
-    "**/*.skip.*",
-    "/**/*.skip/*"
-  ]
+  "ignoreTestFiles": ["**/*.skip.*", "/**/*.skip/*"]
```
Prettier 🤷

jakeleboeuf commented on cypress/support/index.js:
```diff
@@ -18,5 +18,11 @@ import './commands';
 import 'cypress-axe';
 import '@testing-library/cypress/add-commands';
 
+// Delete window.fetch on every window load so we can
+// intercept tests using whatwg-fetch
+Cypress.on('window:before:load', win => {
+  delete win.fetch;
+});
+
```
We need to polyfill the `react-geocode` libs `fetch` request in order to intercept/assert the request string. This is probably a good idea anyways to support older browsers. This tells cypress to use that polyfill instead of it's own implementation registry.

magnificode replied:
Off of @racedale's comment, I think that this is where the Netlify build is failing. We may need to check to see if `window` is defined first, before deleting fetch.

jakeleboeuf commented on src/components/Layout.js:
```diff
@@ -5,6 +5,7 @@ import { StaticQuery, graphql } from 'gatsby';
 import ErrorBoundary from './ErrorBoundary';
 import { SkipNavLink, SkipNavContent } from '@reach/skip-nav';
 import Footer from './footer/Footer';
+import 'whatwg-fetch';
```
https://github.com/Rebuild-Black-Business/RBB-Website/pull/294/files#r447152411

magnificode commented:
@racedale - Any input here sir? I ran through the code here myself and things look pretty solid!

racedale commented:
@magnificode yep looking good to me too!

jakeleboeuf commented:
I’ll track this down and find a solution in the morning 🤙

jakeleboeuf commented:
> Seems to tend to be a gatsby build issue with `whatwg-fetch`
> [mongodb/stitch-js-sdk#381](https://github.com/mongodb/stitch-js-sdk/issues/381)
> 
> Wondering if there's an easier way to solve this?

I've removed the `import 'whatwg-fetch';` statement from the app (and moved it to import the polyfill during the cypress setup) which I _think_ will resolve the Gatsby build issue. I'm not able to reproduce when running `npm build` locally, so kinda shooting in the dark 🤦
